pkg/operator/target_config_reconciler.go

```diff
@@ -171,148 +178,119 @@ func (c TargetConfigReconciler) sync() error {
 		return err
 	}
 
-	if cm, _, err := c.manageConfigMap(kueue); err != nil {
+	resourceVersion := "0"
+	cm, _, err := c.manageConfigMap(kueue)
+	if err != nil {
 		return err
-	} else {
-		resourceVersion := "0"
-		if cm != nil { // SyncConfigMap can return nil
-			resourceVersion = cm.ObjectMeta.ResourceVersion
-		}
-		specAnnotations["kueue/configmap"] = resourceVersion
-
-	}
-
-	crdAnnotations, crdErr := c.manageCustomResources(kueue)
-
-	if crdErr != nil {
-		klog.Error("unable to manage custom resource")
-		return crdErr
 	}
-
-	for key, val := range crdAnnotations {
-		specAnnotations[key] = val
+	if cm != nil {
```

Comment:
cm is always non-nil, right?

Reply:
The code in library-go does return nil in some cases so I don't think we can make the assumption that it always be non-nil.

Reply:
Ok. It's probably because of this block where the err is not handled. We can keep the nil checks.

https://github.com/openshift/library-go/blob/c917e6d528b2d61b1702a36e9d614f9088c95f62/pkg/operator/resource/resourceapply/core.go#L344-L356